### PR TITLE
CMakeLists.txt: don't force the build of a shared library

### PR DESCRIPTION
**Pull request by tpetazzoni**
Building a shared library doesn't work on all platforms, so instead,
let CMake rely on the standard BUILD_SHARED_LIBS variable to decide
whether a static or shared library should be built.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -23,17 +23,18 @@ set(CGI_SRC
...
 	SOVERSION	${PROJECT_VERSION_MAJOR}
 	VERSION		${PROJECT_VERSION}
 )
 
 # install binary
-install(TARGETS ${PROJECT_NAME}-shared
+install(TARGETS ${PROJECT_NAME}
 	EXPORT ${PROJECT_NAME}-targets
 	LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+	ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
```
Comment (LeSpocky) [inline src/CMakeLists.txt]: How is that related to the BUILD_SHARED_LIBS variable change?

Comment (tpetazzoni) [reply src/CMakeLists.txt]: If BUILD_SHARED_LIBS is off, a static library is built, and cmake uses the "ARCHIVE DESTINATION" to know where to install it, while the "LIBRARY DESTINATION" is used when a shared library is built.

**Comment by LeSpocky** [comment]
If I would add an option in CMakeLists.txt to set BUILD_SHARED_LIBS to ON by default, this would still allow to let buildroot or any other build system or user to set it to OFF, right? Then I could take this PR, add this option in another commit and keep the previous behaviour.
